c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -484,14 +553,18 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
             Rvalue::Repeat(ref op, _) => {
                 self.enter_rvalue_operand(0, |v| v.visit_operand(op, None));
             }
-            Rvalue::Ref(_rg, _kind, pl) => {
-                self.enter_rvalue_place(0, |v| v.visit_place(pl));
+            Rvalue::Ref(_rg, kind, pl) => {
+                let mutbl = match kind {
```

Comment:
```suggestion
                let mutbl = matches!(kind, BorrowKind::Mut { .. });
```

Reply:
Writing out all the cases ensures that if a new variant appears in a future rustc version, we'll get a compile error and remember to update this match, instead of potentially having it silently do the wrong thing.